src/java/org/apache/cassandra/index/sai/disk/format/IndexDescriptor.java

```diff
@@ -103,6 +103,11 @@ public static IndexDescriptor create(SSTableReader sstable)
                                    sstable.metadata().comparator);
     }
 
+    public boolean hasClustering()
+    {
+        return clusteringComparator != null && clusteringComparator.size() > 0;
```

Comment:
Can `clusteringComparator` be null?

Reply:
Second this...I don't think there's currently any way it can be, after looking at all the callers.